### PR TITLE
[Model] Add Krea-2 Raw/Turbo support

### PR DESCRIPTION
**Pull request by jjuvonen-amd**
# Summary

Add parallel inference support for [Krea-2-Raw](https://huggingface.co/krea/Krea-2-Raw) and [Krea-2-Turbo](https://huggingface.co/krea/Krea-2-Turbo). Krea-2 is a high-resolution (up to 2048 x 2048) text-to-image model built on a 28-block single-stream transformer architecture. It uses a Qwen3-VL-4B-Instruct text encoder, a 16-channel VAE with 8× spatial compression, and flow-matching scheduling. Text and image tokens are concatenated into a single sequence (Flux-style) before the transformer blocks.

Support added for two models:

Checkpoint | Steps | Guidance | Notes
-- | -- | -- | --
krea/krea-2-raw | 52 | 3.5 | Undistilled base
krea/krea-2-turbo | 8 | - | CFG-free distilled

Supported task: text-to-image only.

## Features

Capability | Supported | Notes
-- | -- | --
Ulysses SP | ✅ | Valid degrees: 1, 2, 3, 4, 6, 8, 12, 16, 24 (divisors of 48)
Ring attention | ❌ | Not supported
Hybrid USP (Ulysses × Ring) | ❌ | Ring not supported
Data parallelism | ✅ |  
FP8 GEMM | ✅ |  
FP4 GEMM | ✅ |  
FSDP | ✅ |  
VAE tiling / slicing | ✅ |  
torch.compile | ✅ |  
Parallel VAE | ❌ | Not supported
PipeFusion | ❌ | Not applicable
CFG parallelism | ❌ | Turbo has no CFG; skipped for Raw

## Examples

`xdit --model krea/krea-2-turbo --ulysses_degree 8 --width 2048 --height 2048 --prompt "a fox walking in the snow" --use_torch_compile`
krea2_turbo_u1r1_tc_True_2048x2048_0

`xdit --model krea/krea-2-raw --ulysses_degree 8 --width 2048 --height 2048 --prompt "a fox walking in the snow" --negative_prompt "bad quality, illustration" --use_torch_compile`
krea2_raw_u1r1_tc_True_2048x2048_0

## Other

Requires diffusers >= 0.39.0

### CODE REVIEW

File: xfuser/model_executor/models/transformers/transformer_krea2.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    [
+                        encoder_attention_mask[:, : text_projected.shape[1]],
+                        encoder_attention_mask.new_ones(B, image_projected.shape[1]),
+                    ],
+                    dim=1,
+                )
+                if pad_len > 0:
+                    combined = torch.cat([combined, combined.new_zeros(B, pad_len)], dim=1)
+                block_attn_mask = make_attn_mask_with_meta(combined)
+                self._attn_mask_cache = (ptr, shape, block_attn_mask)
```
[the remark this answers is not in the document]
Comment (jjuvonen-amd) [reply xfuser/model_executor/models/transformers/transformer_krea2.py:144]: This is intentional and follows the same pattern on how other diffusion models (i.e. Wan or Flux models) handle SP pad tokens. The zero-padding from SP has negligible impact on attention computation and skipping attention mask has performance gains.

**Review by jcaraban — APPROVED** [review]
